File: Day-13/3_Longest_Substring _Without_Repeating_Characters.java
import java.util.ArrayList;

class Anshu{
    static int lengthOfLongestSubstring(String s) {
       
        ArrayList <Character> al = new ArrayList<>();
                         System.out.println(al.size());

        int max = 0;
        for(int i=0 ; i<s.length() ; i++){
            if(al.contains(s.charAt(i))){
                if(al.size()>max){
                    max = al.size();
                }
                al.clear();
               
            }
            al.add(s.charAt(i));
        }
        return max;
    }
    public static void main(String[] args) {
        
        String s1 = "abcabcbb";
        String s2 = "bbbbb";
        String s3 = "pwwkew";

        System.out.println("-- test case 1 --");
        System.out.println("input : "+s1+"\nOutput : "+lengthOfLongestSubstring(s1));
        System.out.println("-- test case 2 --");
        System.out.println("input : "+s2+"\nOutput : "+lengthOfLongestSubstring(s2));
        System.out.println("-- test case 3 --");
        System.out.println("input : "+s3+"\nOutput : "+lengthOfLongestSubstring(s3));
    }
}

File: Day-13/3_Longest_Substring _Without_Repeating_Characters.py
def lengthOfLongestSubstring(s):
    ls = []
    max = 0
    for i in s:
        if i in ls:
            if len(ls)>max:
                max=len(ls)
            ls.clear()
        ls.append(i)
    return max

s1 = "abcabcbb"
s2 = "bbbbb"
s3 = "pwwkew"

print("-- test case 1 --")
print(f"input : {s1}\nOutput : {lengthOfLongestSubstring(s1)}")
print("-- test case 2 --")
print(f"input : {s2}\nOutput : {lengthOfLongestSubstring(s2)}")
print("-- test case 3 --")
print(f"input : {s3}\nOutput : {lengthOfLongestSubstring(s3)}")